### ISSUE TITLE
Ignorované kategorie a více kategorií u produktu

### ISSUE THREAD

**Issue by pavelevap** [pavelevap/ceske-sluzby#101]
A máme tu pokračování #2.

http://www.separatista.net/forum/tema/ceske-sluzby-0-5-ignorovani-kategorie

- [ ] Pokud je produkt zařazen do více kategorií, tak musíme nějak vyřešit, která se bude ve feedu zobrazovat?
- [ ] Možnost ignorovat některé kategorie (např. "Top produkty"), ale pokud je produkt zařazen do nějaké jiné kategorie, tak ho zobrazovat ve feedu (druhou možností je zcela ho ignorovat, patrně to bude muset být volitelné).

**Comment by novetrendy** [comment]
Bod 1. Custom field u produktu - ceske_sluzby_xml_heureka_kategorie - bude mít největší váhu. Pokud nebude vyplněn, použije se z kategorie.

**Comment by pavelevap** [comment]
@novetrendy: Ano, a co když je produkt zařazen do více kategorií najednou (poměrně běžné) a nemá vyplněn custom field? To je vlastně až třetí bod :-) Tento problém ale souvisí hlavně s tím, že má někdo kategorii "Nejprodávanější", kterou přiřazuje libovolným produktům (které mají i další kategorii), a tuto kategorii nechce zobrazovat ve feedu (ale chce tam mít její produkty).

**Comment by pavelevap** [comment]
Jaké mohou nastat varianty?

- Category1
- Category2 - Subcategory
- Category3

Ignorování kategorie je zaškrtnuto (tučně) a produkt je zařazen následovně:

- [x] **Category1** - Všechny produkty z této kategorie jsou vynechány a nedostanou se do feedu.

- [ ] **Category2** - Všechny produkty z této kategorie jsou vynechány a nedostanou se do feedu. Produkty ze Subcategory to ale neovlivní a budou ve feedu. Do budoucna by mělo být možné informace volitelně dědit i pro související podkategorie: #54.

- [ ] **Subcategory** - Asi by měly být vynechány všechny produkty z této podkategorie (a nikoli z nadřazené Category2), ale neověřoval jsem.

- [x] **Category1** a Category3 - Všechny produkty, které jsou zařazeny v kategorii Category1 jsou vynechány a nedostanou se do feedu, i když jsou i v jiné kategorii Category3.

- [ ] **Category1** a Category3 - Co kdyby někdo chtěl vynechat pouze produkty, které jsou zařazeny pouze do jedné kategorie Category1? A produkt, který je i v další kategorii Category3 by se tak měl zobrazovat? V tomto případě by nešlo o vyřazení produktů z této kategorie, ale spíše o ignorování kategorie Category1 (a to ze všech funkcí, kterými produkt v rámci přípravy pro XML feed prochází). Je na to třeba nějaké speciální nastavení nebo by to mělo takto fungovat automaticky? A co když by to chtěl mít někdo jen u nějaké kategorie a u další by se to zase mělo chovat jinak (viz výše)?

- [ ] **Category1** a Subcategory nebo **Category1** a Category2 nebo Category1 a **Subcategory**  - A co kombinace označených kategorií a hierarchie, nebude tam nějaký skrytý problém?

- [ ] Category 1 a Category 3 - Která kategorie bude zobrazena ve feedu, když jich bude u produktu více a žádná zaškrtnutá jako vynechaná? A co kombinace Category1 a Category2, Category1 a Subcategory, Category2 a Subcategory? Nebude nutné si nastavit primární kategorii?

Možná budou ještě nějaké další kombinace :-)

**Comment by pavelevap** [comment]
Další podobný problém: https://www.separatista.net/forum/tema/ceske-sluzby-rozdiel-v-pocte-exportovanych-produktov
